### PR TITLE
detect: strip_pseudo_headers transform

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6546

Describe changes:
- detect: strip_pseudo_headers transform

https://github.com/OISF/suricata-verify/pull/1534
```
SV_BRANCH=pr/1534
```

### CODE REVIEW

File: src/detect-transform-strip-pseudo-headers.c [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                pseudo = false;
+                continue;
+            }
+        }
+        if (!pseudo) {
+            output[j] = input[i];
+            j++;
+        }
+    }
+    InspectionBufferCopy(buffer, output, j);
```
Comment (victorjulien) [inline src/detect-transform-strip-pseudo-headers.c:85]: I think the examples I've see have the `:pseudo-headers` all at the start of the buffer. Wonder if we can have a fast path w/o copy, but where we just have a pointer to the first header after the pseudo headers? Not sure if the inspection buffer logic makes this possible

Comment (catenacyber) [reply src/detect-transform-strip-pseudo-headers.c:85]: > I think the examples I've see have the :pseudo-headers all at the start of the buffer.

This is not guaranteed in all cases...
Do you still want this optimization ?

Comment (victorjulien) [reply src/detect-transform-strip-pseudo-headers.c:85]: Guess we need to profile before adding complexity.

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 17013

File: src/detect-transform-strip-pseudo-headers.c (hunk 1 above)
Comment (catenacyber) [reply src/detect-transform-strip-pseudo-headers.c:85]: So, we can take this version, and optimize if need arises, right ?

Comment (victorjulien) [reply src/detect-transform-strip-pseudo-headers.c:85]: We can optimize later.

**Review by victorjulien — APPROVED** [review]

**Comment by victorjulien** [comment]
Merged in #10167, thanks!

**Pull request closed without merging**